src/resolvers/helpers/__tests__/filter-test.ts

```diff
@@ -90,7 +90,7 @@ describe('Resolver helper `filter` ->', () => {
       const itc = args.filter.type as InputTypeComposer;
       expect(itc.hasField('_id')).toBe(true);
       expect(itc.hasField('name')).toBe(true);
-      expect(itc.hasField('age')).toBe(false);
+      expect(itc.hasField('age')).toBe(true);
```

Comment:
Ouch, tricky case!

If client makes query with `name` and `age` then YES this mongo query will be fast.

But if be requested only `age` then NO this query will be slow, because this field on the [second place of compound index](https://github.com/graphql-compose/graphql-compose-mongoose/blob/aff45d1503a777a4ce7e07836e7acbacc15938fd/src/__mocks__/userModel.ts#L142).

And I don't know what behavior is better...

Reply:
Compound index will only be used if both fields request together, so it will only work on `sort`

On normal query with one field ( name or age ) the compound index will not be used.

So the correct solution will be actually no filters if no singular index for the field.

Reply:
So I suggest either keep them both or remove them both.

Currently they are both there which i think it's fine.

Reply:
> Compound index will only be used if both fields request together

Sorry, but this statement is incorrect. Mongodb can use fields partially from compound index (from left to right). 

<img width="800" alt="Screenshot 2023-11-06 at 10 12 13" src="https://github.com/graphql-compose/graphql-compose-mongoose/assets/1946920/084414ca-214b-4d97-894c-051e09680812">

https://www.mongodb.com/docs/manual/core/indexes/index-types/index-compound/

So for compound index `({name:1, age: 1})` a query with filter:
- `({ name: 'ABC', age: 33 })` will use this index
- `({ name: 'ABC' })` will use this index
- `({ age: 33 })` will NOT use this index

So `name` can be present in `filter` GraphQL type at any case. But `age` is tricky because it will work fast only if client also requested a `name`. And it's sad that with GraphQL is quite difficult to construct such conditional types.

Anyway, I think we can keep your change and publish under major version.

Reply:
Oh yea thank you for highlighting it! 

I could make the first field of the compound index available if you prefer.

It also maybe worth mention this in the readme to make sure users understand index behavior and how best they get the queries.